astropy/config/configuration.py

```diff
@@ -327,13 +317,22 @@ def __init__(
         else:
             self.aliases = aliases
 
+    def __set_name__(self, owner, name):
+        self.name = name
+
     def __set__(self, obj, value):
         return self.set(value)
 
     def __get__(self, obj, objtype=None):
         if obj is None:
             return self
-        return self()
+        # cache value on the descriptor itself, to avoid repeated accesses
+        # to the ConfigObj object which is much slower
+        try:
+            val = self.value
+        except AttributeError:
+            val = self.value = self()
+        return val
```

Comment:
What about this?

```suggestion
        if not hasattr(self, "value"):
            self.value = self()
        return self.value
```

Reply:
`try/except` has no cost if there is no exception, so I think it is good to use it (`hasattr` uses it internally, but now you do have the cost of the `if` statement). I'd try to shave off another few ns with
```
try:
    return self.value
except AttributeError:
    self.value = val = self()
    return val
```

Reply:
yeah try/except is commonly used in such cases both for perf and also `hasattr` can have some side effects so simpler is better here.